### PR TITLE
feat(GKE): add terraform support for dataplaneV2 & CalicoNetworkPolicy in clusterNetwork config

### PR DESCRIPTION
**Pull request by sharan-rafay**
Jira: https://rafaysystems.atlassian.net/browse/RC-34510

### CODE REVIEW

**Review by poornima-rafay — APPROVED** [review]

File: docs/resources/gke_cluster.md [hunk 1]
```diff
@@ -263,6 +265,8 @@ For ZonalCluster only zone information is sufficient. For Regional Cluster, both
 - `service_address_range` (String) Cluster services will be assigned an IP address from this IP address range. Enter a range (in CIDR notation) within a network range, a mask, or leave this field blank to use a default range.
 - `pod_secondary_range_name` (String) Cluster pods are assigned an IP from the selected node subnet's secondary CIDR address range.
 - `service_secondary_range_name` (String) Cluster services are assigned an IP from the selected node subnetes secondary CIDR address range.
+- `data_plane_V2` (String) Dataplane V2 is optimized for Kubernetes networking which is implemented using eBPF.
```
Comment (akshay196-rafay) [inline docs/resources/gke_cluster.md:268]: List supported values.

File: docs/resources/gke_cluster.md [hunk 2]
```diff
@@ -263,6 +265,8 @@ For ZonalCluster only zone information is sufficient. For Regional Cluster, both
...
+- `calico_network_policy` (String) The Kubernetes Network Policy API allows the cluster administrator to specify what pods are allowed to communicate with each other. Cannot use with `data_plane_V2` config
```
Comment (akshay196-rafay) [inline docs/resources/gke_cluster.md:269]: List supported values.

**Review by akshay196-rafay — CHANGES_REQUESTED** [review]

File: docs/resources/gke_cluster.md (hunk 2 above)
Comment (sharan-rafay) [reply docs/resources/gke_cluster.md:269]: updated.

File: docs/resources/gke_cluster.md (hunk 1 above)
Comment (sharan-rafay) [reply docs/resources/gke_cluster.md:268]: updated

**Review by akshay196-rafay — APPROVED** [review]

**Pull request merged**